internalshared/configutil/request_limiter.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package configutil
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/ast"
+)
+
+type RequestLimiter struct {
+	UnusedKeys UnusedKeyMap `hcl:",unusedKeyPositions"`
+
+	Disable    bool        `hcl:"-"`
+	DisableRaw interface{} `hcl:"disable"`
+}
+
+func (r *RequestLimiter) Validate(source string) []ConfigError {
+	return ValidateUnusedFields(r.UnusedKeys, source)
+}
+
+func (r *RequestLimiter) GoString() string {
```

Comment:
<img src="https://media3.giphy.com/media/JaNtIG4UnKzD2/giphy.gif"/>

What's this for?

Reply:
Maybe for tests? Truthfully not sure, but other configs used this so I followed suit.